### PR TITLE
Handle errors when registering new queue readers

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Handle errors when registering new queue readers
- Some refactoring in queue monitor & mitigator & action to trigger back pressure when registering error happens.

**Why?**
- Error handling

**How did you test it?**
- Add unit test

**Potential risks**
- N/A, those code path should never be executed.

**Is hotfix candidate?**
- No

### CODE REVIEW

File: service/history/queues/action_pending_task_count.go [hunk 1]
```diff
@@ -216,17 +221,49 @@ func (a *actionQueuePendingTask) splitAndClearSlice(
...
 
 		nextReader, ok := readerGroup.ReaderByID(readerID + 1)
-		if ok {
-			nextReader.MergeSlices(splitSlices...)
-		} else {
-			nextReader = readerGroup.NewReader(readerID+1, splitSlices...)
+		if !ok {
+			// this should never happen, we already ensured all readers are created.
+			// we have no choice but to put those slices back
+			reader.MergeSlices(splitSlices...)
```
Comment (yiminc) [inline service/history/queues/action_pending_task_count.go:227]: write some noticeable logs, or even emit metric here?

File: service/history/queues/action_pending_task_count.go [hunk 2]
```diff
@@ -39,12 +39,13 @@ const (
 	clearSliceThrottleDuration = 10 * time.Second
 )
 
+var _ Action = (*actionQueuePendingTask)(nil)
```
Comment (yiminc) [inline service/history/queues/action_pending_task_count.go:42]: please add comment to explain what this action is about, what alert is it react to, and how does it mitigate the alert.

**Review by yiminc — APPROVED** [review]
Do we have test coverage targeting to every possible alert types we defined and observe the action taken that mitigate the problem?

File: service/history/queues/action_pending_task_count.go (hunk 2 above)
Comment (yycptt) [reply service/history/queues/action_pending_task_count.go:42]: yeah I will send follow up PRs to comment and add tests for those actions.

**Pull request merged**